Tests/NIOTests/EventLoopTest.swift

```diff
@@ -928,4 +928,48 @@ public final class EventLoopTest : XCTestCase {
             }
         }
     }
+
+    func testSuccessfulFlatSubmit() {
+        let eventLoop = EmbeddedEventLoop()
+        let future = eventLoop.flatSubmit {
+            eventLoop.makeSucceededFuture(1)
+        }
+        eventLoop.run()
+        XCTAssertNoThrow(XCTAssertEqual(1, try future.wait()))
+    }
+
+    func testFailingFlatSubmit() {
+        enum TestError: Error { case failed }
+
+        let eventLoop = EmbeddedEventLoop()
+        let future = eventLoop.flatSubmit { () -> EventLoopFuture<Int> in
+            eventLoop.makeFailedFuture(TestError.failed)
+        }
+        eventLoop.run()
+        XCTAssertThrowsError(try future.wait()) { error in
+            XCTAssertEqual(.failed, error as? TestError)
+        }
+    }
+
+    func testFlatSubmitOnShutdownLoopGroup() {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        let eventLoop = eventLoopGroup.next()
+        XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
+
+        let _ = eventLoop.flatSubmit { () -> EventLoopFuture<Void> in
```

Comment:
shame that our `submit` API doesn't return a failed future if the ELG is shut down but that's definitely not this PR's responsibility.

Reply:
I had to experiment to determine what the expected behavior was. On my machine, the tests would hang if I called ```wait``` after submitting a task to a shutdown group.

However, I noticed that the tests seem to be failing intermittently on the build server (one failed twice, one failed only once). So it seems that the submitted tasks are somehow running after the group is shutdown? I haven't seen that behavior on my machine.

Reply:
--> 1177

Reply:
Ok, sounds good.